<?xml version="1.0"?>
<launch>
  
  <arg name="geotiff_map_file_path" default="$(find ugv_localization)/map"/>
  <arg name="data" default="test_03"/>


  <!-- play data -->
  <param name="use_sim_time" value="true"/>
  <node pkg="rosbag" type="play" name="rosbag_play" args="$(find ugv_localization)/bag/$(arg data).bag --clock"/>


  <!-- convert GPS into map frame -->
  <node pkg="ugv_localization" type="gps_odom_node" name="gps_odom_node" output="screen">
    <param name="coordinate_system" type="string" value="wgs84"/>
	<param name="topic_gps" type="string" value="/gps/fix"/>
	<param name="topic_odom" type="string" value="/odometry/gps"/>
	<param name="latitude" type="double" value="39.326465"/>
	<param name="longitude" type="double" value="-76.621554"/>
	<param name="frame_id" type="string" value="map"/>
	<param name="child_frame_id" type="string" value="base_link"/>
	<param name="cov_threshold" type="double" value="1"/>
	<param name="cov_scale" type="double" value="100"/>
  </node>


  <!-- publish /world/fix -->
  <include file="$(find ugv_localization)/launch/world_frame_gps_publisher.launch"/>


  <!-- publish required tf (transformation) -->
  <include file="$(find ugv_localization)/launch/static_tfs.launch"/>

  <!-- tunnel_hackerman -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="map2og_org" args="31.6625232865999 43.60933096451932 0 1.3514709226064314 0 0 map og_org"/>

  <!-- tunnel gilman -->
  <!-- node pkg="tf2_ros" type="static_transform_publisher" name="map2og_org" args="32.76622780132711 178.06773404751007 0 1.7363034563005664 0 0 map og_org"/ -->


  <!-- main functionality -->
  <include file="$(find ugv_localization)/launch/hector_mapping.launch"/>
 
  <include file="$(find ugv_localization)/launch/hector_geotiff_mapper.launch">
    <arg name="trajectory_source_frame_name" value="scanmatcher_frame"/>
    <arg name="map_file_path" value="$(arg geotiff_map_file_path)"/> 
  </include>


  <!-- rviz -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find ugv_localization)/rviz/mapping.rviz"/>

</launch>
